File: cpp/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        vector<int> result;
        for (int i = 0; i < nums.size(); ++i) {
            if (numMap.find(target - nums[i]) != numMap.end()) {
                result = {numMap[target - nums[i]], i};
            } else {
                numMap[nums[i]] = i;
            }
        }

        return result;
    }
};

File: js/1_Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numsSwap = {};
    let result = [-1, -1];
    nums.forEach((element, index) => {
        if (numsSwap.hasOwnProperty(target - element)) {
            result = [numsSwap[target - element], index];
        } else {
            numsSwap[element] = index;
        }
    });
    return result;
};
